### PR TITLE
Validate domain-linkage URL making sure they only include an origin

### PR DESCRIPTION
**Pull request by UMR1352**
# Description of change
See issue #1265

## Links to any relevant issues
Fixes issue #1265 

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested
Added a new test that makes sure only URLs with no path, no query and no fragment are accepted

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by UMR1352** [comment]
@abdulmth I'm not exactly satisfied by these changes, but I wanted to ask your opinion first. What I'd like to do is extract the URL validation logic used here into a new type `DomainLinkageOrigin` and use that instead of an `Url`. This could definitely be more type-safe and just as ergonomic as `Url` if provided with the right interface. What do you say?

File: identity_credential/src/domain_linkage/domain_linkage_configuration.rs [hunk 1]
```diff
@@ -128,6 +128,9 @@ mod __fetch_configuration {
       if domain.scheme() != "https" {
         return Err(DomainLinkageError("domain` does not use `https` protocol".into()));
       }
+      if domain.path() != "/" || domain.query().is_some() || domain.fragment().is_some() {
+        return Err(DomainLinkageError("domain mut not include any path, query of fragment".into()));
```
Comment (abdulmth) [inline identity_credential/src/domain_linkage/domain_linkage_configuration.rs:132]:
```suggestion
        return Err(DomainLinkageError("domain must not include any path, query or fragment".into()));
```

File: identity_credential/src/domain_linkage/domain_linkage_credential_builder.rs [hunk 2]
```diff
@@ -139,6 +144,18 @@ mod tests {
...
     assert!(matches!(err, Error::DomainLinkageError(_)));
   }
+  #[test]
+  fn test_builder_origin_is_a_url() {
+    let issuer: CoreDID = "did:example:issuer".parse().unwrap();
+    let err: Error = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .origin(Url::parse("https://example.com/foo?bar=420#baz").unwrap())
```
Comment (abdulmth) [inline identity_credential/src/domain_linkage/domain_linkage_credential_builder.rs:154]: I would add other test cases for `"https://example.com/?bar=420#baz"` and `"https://example.com/#baz"`

File: identity_credential/src/domain_linkage/domain_linkage_configuration.rs [hunk 3]
```diff
@@ -128,6 +128,9 @@ mod __fetch_configuration {
       if domain.scheme() != "https" {
         return Err(DomainLinkageError("domain` does not use `https` protocol".into()));
       }
+      if domain.path() != "/" || domain.query().is_some() || domain.fragment().is_some() {
```
Comment (abdulmth) [inline identity_credential/src/domain_linkage/domain_linkage_configuration.rs:131]: Not sure about the part `domain.path() != "/"` would it return false if the URL was `http://www.example.com/` and not `http://www.example.com` ?  we can probably also add positive tests for these cases too.

**Review by abdulmth — COMMENTED** [review]
I think adding a new type for internal use is a bit too much (exposing it would be a breaking change, no?). We can just add a function like `isOriginOnly` or something similar to avoid the duplication, but it looks good otherwise.

File: identity_credential/src/domain_linkage/utils.rs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+use identity_core::common::Url;
```
Comment (abdulmth) [inline identity_credential/src/domain_linkage/utils.rs:1]:
```suggestion
// Copyright 2020-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use identity_core::common::Url;
```

**Review by abdulmth — COMMENTED** [review]
I think one more place to add this check is `LinkedDomainService::check_structure()` since the  endpoints must also be origins only otherwise the service is invalid.

**Review by abdulmth — APPROVED** [review]

**Pull request merged**